tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -89,202 +97,229 @@ type Data = {
 
 export const PersonalCooking = (props, context) => {
   const { act, data } = useBackend<Data>(context);
+  const { busy, display_compact, display_craftable_only, craftability, diet } =
+    data;
   const [searchText, setSearchText] = useLocalState(context, 'searchText', '');
+  const [pages, setPages] = useLocalState(context, 'pages', 1);
   const [activeCategory, setCategory] = useLocalState(
     context,
     'category',
-    data.categories[0]
+    Object.keys(craftability).length ? 'Can Make' : data.categories[0]
   );
   const [activeType, setType] = useLocalState(
     context,
     'foodtype',
-    data.foodtypes[0]
+    Object.keys(craftability).length ? 'Can Make' : data.foodtypes[0]
   );
   const [typeMode, setTypeMode] = useLocalState(context, 'typeMode', false);
   const searchName = createSearch(searchText, (item: Recipe) => item.name);
-  const craftabilityByRef = data.craftability;
   let recipes = flow([
     filter<Recipe>(
       (recipe) =>
         // If craftable only is selected, then filter by craftability
-        (!data.display_craftable_only ||
-          Boolean(craftabilityByRef[recipe.ref])) &&
+        (!display_craftable_only || Boolean(craftability[recipe.ref])) &&
         // Ignore categories and types when searching
         (searchText.length > 0 ||
           // Is type mode and the active type matches
           (typeMode &&
-            (activeType === '' || recipe.foodtypes?.includes(activeType))) ||
+            ((activeType === 'Can Make' && Boolean(craftability[recipe.ref])) ||
+              recipe.foodtypes?.includes(activeType))) ||
           // Is category mode and the active categroy matches
           (!typeMode &&
-            (activeCategory === '' || recipe.category === activeCategory)))
+            ((activeCategory === 'Can Make' &&
+              Boolean(craftability[recipe.ref])) ||
+              recipe.category === activeCategory)))
     ),
     sortBy<Recipe>((recipe) => [
-      -Number(craftabilityByRef[recipe.ref]),
-      recipe.name,
+      -Number(craftability[recipe.ref]),
+      recipe.name.toLocaleLowerCase(),
```

Comment:
Last thing - locale aware functions are very slow in IE11, and you may inadvertedly make your UI much slower. I did the same back in the day with a Cargo UI, which got 3x slower from localeCompare in the sort.

Reply:
How did you get it? Is there a way to profile dom rendering and the request payload?